File: src/chipschallenge/trapreleasebehaviors/DefaultTrapReleaseBehavior.java
package chipschallenge.trapreleasebehaviors;

import chipschallenge.Block;

public class DefaultTrapReleaseBehavior implements TrapReleaseBehavior {

    private DefaultTrapReleaseBehavior() {
    }
    private static DefaultTrapReleaseBehavior mInstance = null;

    public static synchronized DefaultTrapReleaseBehavior getInstance() {
        if (mInstance == null) {
            mInstance = new DefaultTrapReleaseBehavior();
        }
        return mInstance;
    }

    public void releaseFromTrap(Block trapped) {
        trapped.setTrapped(false);
    }
}

File: ChipsChallenge.Shared/Trapreleasebehaviors/DefaultTrapReleaseBehavior.cs
namespace ChipsChallenge.Shared.Trapreleasebehaviors
{
    public class DefaultTrapReleaseBehavior : ITrapReleaseBehavior
    {
        private static DefaultTrapReleaseBehavior instance;

        private DefaultTrapReleaseBehavior()
        {
        }

        public static DefaultTrapReleaseBehavior Instance
        {
            get
            {
                lock (typeof(DefaultTrapReleaseBehavior))
                {
                    return instance ?? (instance = new DefaultTrapReleaseBehavior());
                }
            }
        }

        public virtual void ReleaseFromTrap(Block trapped)
        {
            trapped.Trapped = false;
        }
    }
}